sql/sql_parse.cc

```diff
@@ -8389,7 +8389,11 @@ TABLE_LIST *st_select_lex::add_table_to_list(THD *thd,
     DBUG_RETURN(0);
   else
     fqtn= FALSE;
-  bool info_schema= is_infoschema_db(&db);
+  /*
+    The test for length==0 is for performance purposes as is_infoschema_db()
+    is quite heavy. Without explict db SQL syntax, length==0 is a common case.
+  */
+  bool info_schema= db.length ? is_infoschema_db(&db) : false;
```

Comment:
I don't understand. What is quite heavy? `is_infoschema_db` is
```c++
inline bool is_infoschema_db(const LEX_CSTRING *name)
{
  return lex_string_eq(&INFORMATION_SCHEMA_NAME, name);
}
```
and `lex_string_eq` is
```c++
static inline bool lex_string_eq(const LEX_CSTRING *a, const LEX_CSTRING *b)
{
  if (a->length != b->length)
    return 0;                                   /* Different */
  return strcasecmp(a->str, b->str) == 0;
}
```
note that everything is inlined

Reply:
Ok, seems like 11.8 is when this changed, which is already fixed. Thanks for the review - closing this aspect of the fix.